File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        vector<int> ans(2,0);
        int l = 0, r = numbers.size() - 1;
        int sum;
        while (l < r) {
            sum = numbers[l] + numbers[r];
            if (sum == target) {
                ans[0] = l+1;
                ans[1] = r+1;
                break;
            }
            if (sum < target) {
                l++;
            }
            if (sum > target) {
                r--;
            }
        }
        return ans;

    }
};

File: Data Structures & Algorithms/search-2d-matrix/submission-1.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        l = 0
        m = len(matrix) #3
        n = len(matrix[0]) #4
        r =  m * n - 1

        while l <= r:
            mid = int(l + (r - l) / 2)
            c_l = mid // n
            c_r = mid % n
            print(f"{l} {r} {mid} {c_l} {c_r}")
            if matrix[c_l][c_r] == target:
                return True
            elif matrix[c_l][c_r] > target:
                r = mid - 1
            else:
                l = mid + 1
        return False 